internal/tofu/context_plugins.go

```diff
@@ -77,6 +113,7 @@ func (cp *contextPlugins) ProviderSchema(addr addrs.Provider) (providers.Provide
 	// That is because we're checking *prior* to the provider's instantiation.
 	// GetProviderSchemaOptional only says that *if we instantiate a provider*,
 	// then we need to run the get schema call at least once.
+	// BUG This SHORT CIRCUITS the logic below and is not the only code which inserts provider schemas into the cache!!
```

Comment:
do we need to do something about this?

Reply:
Not at this point, it's more so a marker for when we refactor how providers are handled.